pkg/julia/manifest/parse.go

```diff
@@ -0,0 +1,159 @@
+package julia
+
+import (
+	"io"
+	"sort"
+
+	"github.com/BurntSushi/toml"
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+
+	"golang.org/x/exp/maps"
+	"golang.org/x/xerrors"
+)
+
+type primitiveManifest struct {
+	JuliaVersion   string                           `toml:"julia_version"`
+	ManifestFormat string                           `toml:"manifest_format"`
+	Dependencies   map[string][]primitiveDependency `toml:"deps"` // e.g. [[deps.Foo]]
+}
+
+type primitiveDependency struct {
+	Dependencies toml.Primitive `toml:"deps"` // by name. e.g. deps = ["Foo"] or [deps.Foo.deps]
+	UUID         string         `toml:"uuid"`
+	Version      string         `toml:"version"` // not specified for stdlib packages, which are of the Julia version
+	DependsOn    []string       `toml:"-"`       // list of dependent UUID's.
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var oldDeps map[string][]primitiveDependency
+	var primMan primitiveManifest
+	var manMetadata toml.MetaData
+	decoder := toml.NewDecoder(r)
+	// Try to read the old Manifest format. If that fails, try the new format.
+	if _, err := decoder.Decode(&oldDeps); err != nil {
```

Comment:
Looks like we need to return `manMetadata` here too.

Reply:
No need, because the old deps format can always be parsed into that object. We only need the parse metadata for the new manifest format.

Reply:
okay, thanks for this explain!